### ISSUE TITLE
Clean up our content-type tests

### ISSUE THREAD

**Issue by XhmikosR** [MaxCDN/bootstrapcdn#1032]
From #1023:

1. We need to remove `content-type: undefined` if we are going to check individually later
2. We don't need to run `assertItWorks` both for each group test and in `public/**/*.*`
3. There shouldn't be any unknown file types so we should remove the `typeof helpers.CONTENT_TYPE_MAP[ext] === 'undefined'` check
4. We probably should remove the custom code that stops testing `twitter-bootstrap` now that tests are fast
5. `content-type` in `public/**/*.*` is duplicate for most files since we do it individually too

I suggest that

1. we stop running functional explicitly for each group of files
2. change the `assertHeader` some way so that the `content-type` is tested

So basically, in functional test `public/**/*.*` with `assertItWorks` and then with `assertHeader` assuming we manage to adapt `assertHeader` to check for the proper `content-type`.

**Comment by XhmikosR** [comment]
I think this is sort of addressed now.
